File: workplace2/src/learning_service/scripts/turtle_spawn.py
# !/usr/bin/env python
# _*_ coding: utf-8 _*_

import sys
import rospy
from turtlesim.srv import Spawn

def turtle_spawn():
    rospy.init_node('turtle_spawn')

    rospy.wait_for_service('/spawn')
    try:
        add_turtle = rospy.ServiceProxy('/spawn', Spawn)

        response = add_turtle(2.0, 2.0, 0.0, "turtle2")
        return response.name
    except rospy.ServiceException,e:
        print "Service call failed: %s"%e

if _name_ == "_main_":
    print "Spwan turtle successfully [name:%s]" %(turtle_spawn())

File: workplace2/src/learning_service/src/turtle_spawn.cpp
/*************************************************************************
	> File Name: turtle_spawn.cpp
	> Author: 
	> Mail: 
	> Created Time: 2020年11月09日 星期一 15时17分29秒
 ************************************************************************/

#include <ros/ros.h>
#include <turtlesim/Spawn.h>

int main(int argc, char **argv)
{
    ros::init(argc, argv, "turtle_spawn");

    ros::NodeHandle node;

    ros::service::waitForService("/spawn");//jaming function like wait();
    ros::ServiceClient add_turtle = node.serviceClient<turtlesim::Spawn>("/spawn");

    turtlesim::Spawn srv;
    srv.request.x = 2.0;
    srv.request.y = 2.0;
    srv.request.name = "turtle2";

    ROS_INFO("Call service to spawn turtle[x:%0.6f, y:%0.6f, name:%s]",
             srv.request.x, srv.request.y, srv.request.name.c_str());

    add_turtle.call(srv); //ask for service with data(package) named srv

    ROS_INFO("Spwan turtle successfully [name:%s]", srv.response.name.c_str());

    return 0;
}
